components/App.js

```diff
@@ -1,59 +1,78 @@
-var App = React.createClass({
+const App = React.createClass({
 
     getInitialState() {
         return {
             loading: false,
             searchingText: '',
             gif: {}
-        };
+        }
     },
 
     handleSearch: function(searchingText) {
         this.setState({
             loading: true
         });
 
-        this.getGif(searchingText, function(gif) {
-            this.setState({
-                loading: false,
-                gif: gif,
-                searchingText: searchingText
-            });
-        }.bind(this));
+        this.getGif(searchingText)
+        .then(data => { 
+                        const gif = {
+                              url: data.fixed_width_downsampled_url,
+                              sourceUrl: data.url
+                              };
+
+                        this.setState({
+                            loading: false,
+                            gif: gif,
```

Comment:
kiedy wartość nazywa się tak samo jak klucz możesz użyć skróconej nazwy, w tym wypadku wyglądałoby to tak:
```
this.setState({
  loading: false,
  gif,
  searchingText
})